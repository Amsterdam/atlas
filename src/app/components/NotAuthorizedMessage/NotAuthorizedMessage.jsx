import React from 'react';
import PropTypes from 'prop-types';
import { AngularWrapper } from 'react-angular';

const NotAuthorizedPanel = ({ dataset }) => (
  <AngularWrapper
    moduleName={'dpPanelWrapper'}
    component="dpPanel"
    className="qa-disabled-message"
    dependencies={['atlas']}
    bindings={{
      isPanelVisible: true,
      canClose: true
    }}
    interpolateBindings={{
      type: 'warning'
    }}
  >
    <div>
      <p className="c-panel__paragraph">
        Medewerkers met speciale bevoegdheden kunnen inloggen om kadastrale objecten met zakelijk
        rechthebbenden te bekijken.
      </p>
      <p className="c-panel__paragraph">
        Zie {dataset}
      </p>
    </div>
  </AngularWrapper>
);

NotAuthorizedPanel.propTypes = {
  dataset: PropTypes.string.isRequired
};

export default NotAuthorizedPanel;
