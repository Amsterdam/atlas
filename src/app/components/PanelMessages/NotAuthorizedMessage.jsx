import React from 'react';
import PropTypes from 'prop-types';
import Link from 'redux-first-router-link';
import { routing } from '../../routes';
import Notification from '../../../shared/components/notification/Notification';

const NotAuthorizedMessage = ({ scopeError }) => (
  <Notification type="warning" >
    <div>
      { (scopeError === 'BRK/RSN')
          ? <p className="c-panel__paragraph">
              Medewerkers met speciale bevoegdheden kunnen inloggen om kadastrale objecten met
              zakelijk rechthebbenden te bekijken.
            </p>
          : <p className="c-panel__paragraph">
              Medewerkers/ketenpartners van Gemeente Amsterdam kunnen inloggen om maatschappelijke
              activiteiten en vestigingen te bekijken.
            </p>
      }
      <p className="c-panel__paragraph">
        Zie <Link
          className="c-link--light qa-link-to-page-button qa-dp-link"
          to={{
            type: routing.bediening.type,
            payload: { deeplink: 'inloggen' }
          }}
        >
          Help &#62; Bediening &#62; Inloggen
        </Link>
      </p>
    </div>
  </Notification>
);

NotAuthorizedMessage.defaultProps = {
  scopeError: ''
};

NotAuthorizedMessage.propTypes = {
  scopeError: PropTypes.string
};

export default NotAuthorizedMessage;
