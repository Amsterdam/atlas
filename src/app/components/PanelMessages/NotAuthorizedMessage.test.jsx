import React from 'react';
import { shallow } from 'enzyme';
import NotAuthorizedMessage from './NotAuthorizedMessage';

describe('NotAuthorizedMessage', () => {
  it('should render everything', () => {
    const component = shallow(<NotAuthorizedMessage scopeError />);
    expect(component).toMatchSnapshot();
  });
});
