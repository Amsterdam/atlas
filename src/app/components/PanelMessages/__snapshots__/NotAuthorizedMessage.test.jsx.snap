// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotAuthorizedMessage should render everything 1`] = `
<Notification
  type="warning"
>
  <div>
    <p
      className="c-panel__paragraph"
    >
      Medewerkers/ketenpartners van Gemeente Amsterdam kunnen inloggen om maatschappelijke activiteiten en vestigingen te bekijken. 
    </p>
    <Connect(LoginLink) />
  </div>
</Notification>
`;
